collector/lib/ResourceSelector.cpp

```diff
@@ -0,0 +1,143 @@
+//
```

Comment:
This file already exists in another PR. Some changes were made to it in this PR.